### PR TITLE
feat: minimize on startup

### LINKED ISSUE #45: feat: start minimized
I have rquickshare set to start on login and I'd like to be able to make it start with the window minimized/hidden

### PR DESCRIPTION
**Pull request by tatejuan12**
Added feature to start RQuickShare hidden in taskbar. Also added a toggle that can set the feature on or off. Set on as default, given I believe no one wants to have RQuickShare open up on startup.

This is my very first OS contribution, so please advise if I am doing something wrong!

Developed and tested on Arch Linux, Plasma 6. 

Closes #45

### CODE REVIEW

**Comment by Martichou** [comment]
Hey! Thanks, I'll check that asap but I'm a bit busy currently :)

**Comment by Keleva** [comment]
Hay forma de hacerlo correr al inicio sin abrir la GUI, en gestores de pantalla como awesomeWM o bspwm?, saludos.

File: frontend/src-tauri/src/store.rs [hunk 1]
```diff
@@ -45,6 +47,26 @@ pub fn get_realclose(app_handle: &AppHandle) -> bool {
...
+        ".settings.json",
+        |store| {
+            return Ok(store
+                .get("minimizeonstartup")
+                .and_then(|json| json.as_bool()));
+        },
+    );
+
+    match minimizeonstartup {
+        Ok(r) => r.unwrap_or(false),
```
Comment (Martichou) [inline frontend/src-tauri/src/store.rs:63]: should this return True by default as your intention was to minimize by default?

Comment (Martichou) [reply frontend/src-tauri/src/store.rs:63]: see my general comment too

File: frontend/src-tauri/src/main.rs [hunk 2]
```diff
@@ -277,3 +285,8 @@ fn rs2js_endpointinfo<R: tauri::Runtime>(message: EndpointInfo, manager: &impl M
 fn sanity_check() {
     info!("sanity_check");
 }
+
+#[tauri::command]
```
Comment (Martichou) [inline frontend/src-tauri/src/main.rs:289]: can you move that to a separate file like to other cmds?

File: frontend/src-tauri/tauri.conf.json [hunk 3]
```diff
@@ -50,7 +50,10 @@
       "active": false
     },
     "allowlist": {
-      "all": true
+      "all": true,
+      "window": {
+        "hide": true
+      }
```
Comment (Martichou) [inline frontend/src-tauri/tauri.conf.json:56]: not needed afaik

File: frontend/src/components/HomePage.vue [hunk 4]
```diff
@@ -585,11 +594,23 @@ export default {
...
+			
+			await this.store.set(minimizeonstartupKey, minimizeonstartup);
+			await this.store.save();
+			this.minimizeonstartup = minimizeonstartup;
+		},
+		applyMinimizeOnStartup: async function(){
+			this.minimizeonstartup = await this.store.get(minimizeonstartupKey) ?? false;
+			if (this.minimizeonstartup) await enable();
+			else await disable();
+
```
Comment (Martichou) [inline frontend/src/components/HomePage.vue:613]: remove

File: frontend/src/components/HomePage.vue [hunk 5]
```diff
@@ -585,11 +594,23 @@ export default {
...
-				await disable();
-			}
+			if (this.autostart) await enable();
+			else await disable();
+			
+		},
+		setMinimizeOnStartup: async function(minimizeonstartup: boolean) {
+			if (minimizeonstartup) await enable();
+			else await disable();
+
```
Comment (Martichou) [inline frontend/src/components/HomePage.vue:604]: space

File: frontend/src/components/HomePage.vue [hunk 6]
```diff
@@ -585,11 +594,23 @@ export default {
...
 			this.autostart = await this.store.get(autostartKey) ?? false;
 
-			if (this.autostart) {
-				await enable();
-			} else {
-				await disable();
-			}
+			if (this.autostart) await enable();
+			else await disable();
+
```
Comment (Martichou) [inline frontend/src/components/HomePage.vue:599]: remove

File: frontend/src/components/HomePage.vue [hunk 7]
```diff
@@ -409,11 +417,12 @@ export default {
...
-				await this.setAutostart(true);
-			} else {
-				await this.applyAutostart();
-			}
+			if (!await this.store.has(autostartKey)) await this.setAutostart(true); 
+			else await this.applyAutostart();
+
+			if (!await this.store.has(minimizeonstartupKey)) await this.setMinimizeOnStartup(true);
+			else await this.applyMinimizeOnStartup();
+
```
Comment (Martichou) [inline frontend/src/components/HomePage.vue:425]: remove

**Review by Martichou — CHANGES_REQUESTED** [review]
I'd prefer this to be disabled by default as I may see this cause a small quirk.

If a computer is slow enough, the widow can popup and then disappear  before the minimize_window_on_startup in the setup is called. So until more testers report it working correctly, I'd keep it disabled by default.

**Comment by bruglet** [comment]
How do we test this so it can get merged in?

**Comment by Martichou** [comment]
As there's no update from @functionalities I'll make the requested changes myself once I have some spare time :) after that it should be good to go.

**Comment by Martichou** [comment]
Done here: https://github.com/Martichou/rquickshare/commit/eac7dab0dcd8b609cd021f7f29d1e558dca4472c

**Pull request closed without merging**